viewing_party/party.py

```diff
@@ -0,0 +1,154 @@
+        if movie_data["title"] == title:
+            user_data["watched"].append(movie_data)
+            user_data["watchlist"].remove(movie_data)
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    average_data = []
+    for movie_data in user_data["watched"]:
+        average_data.append(movie_data["rating"])
+
+    if not average_data:
+            return 0
+    else: 
+        return sum(average_data)/len(average_data)
+
+def get_most_watched_genre(user_data):
+    most_genre = {}
+    for movie_data in user_data["watched"]:
+        current_genre = movie_data["genre"]
+        if current_genre in most_genre:
+            most_genre[current_genre] += 1
+        else:
+            most_genre[current_genre] = 1
+    #alternative way
+    # most_watched_num = 0
+    # for key, value in most_genre:
+    #     if value > 0:
+    #         most_watched_num = value
+    
+    # for key, value in most_genre:
+    #     if value == most_watched_num:
+    #         return key 
+    if most_genre:
+        return max(most_genre, key=most_genre.get)
+    else:
+        return None
+
+def get_unique_watched(user_data):
+# user_data will be a dictionary with a "watched" list of movie dictionaries, and a "friends"
+# "friends" is a list, where each item in the list is a dictionary
+# This dictionary has a key "watched", which has a list of movie dictionaries
+# Each movie dirctionary has a "title"
+# Return a list of dictionaries that the user has watched, but none of their friends have watched
+    unique_watched = []
+    unique_watched += user_data["watched"]
+
+    for friend in user_data["friends"]:
+        for movie_data in friend["watched"]:
+            if movie_data in unique_watched:
+                unique_watched.remove(movie_data)
+    return unique_watched
+
+def get_friends_unique_watched(user_data):
+    # Makes a list of what the user's friends have watched that the user has not watched
+    friends_unique_watched = []
+
+    for friend in user_data["friends"]:
+        friends_unique_watched += friend["watched"]
+
+    for user_watched in user_data["watched"]:
+        if user_watched in friends_unique_watched:
+            friends_unique_watched.remove(user_watched)
+
+    true_unique = []
+    for movie in friends_unique_watched:
+        if movie not in true_unique:
+            true_unique.append(movie)
+
+    return true_unique
+
+def get_available_recs(user_data):
+    # Return a list of dictionaries that has "title" key and the "movie title" as value
+    user_watched_titles = [] 
+    rec_movies = []
+
+    for user_movie_data in user_data["watched"]:
+        user_watched_titles.append(user_movie_data["title"])
+
+    for friend in user_data["friends"]:
+        for friend_movie_data in friend["watched"]:
+            if friend_movie_data["title"] not in user_watched_titles:
+                if friend_movie_data["host"] in user_data["subscriptions"]:
+                    if friend_movie_data not in rec_movies:
```

Comment:
Chained `if` statements like this can be combined into one conditional statement where each section is combined with `and`.